.github/workflows/test.yaml

```diff
@@ -0,0 +1,41 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: tests
+
+on:
+  push:
+    branches: [ "main", "develop" ]
+  pull_request:
+    branches: [ "main", "develop" ]
+
+permissions:
+  contents: read
+
+jobs:
+
+  lint_and_test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.10', '3.11']
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v3
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements-dev.txt
+        pip install flake8-bandit flake8-bugbear flake8-builtins flake8-comprehensions flake8-docstrings flake8-eradicate flake8-isort
+    - name: Lint with flake8
+      run: |
+        # stop the build if there are Python syntax errors or undefined names, this will be obsolete when we have pre-commit.ci
+        flake8 . --max-line-length=88 --extend-ignore=E203,I001,I005,D101,D103,D105,B023,S603,S608,S101,S310,E402 --docstring-convention=numpy

```

Comment:
this seems to be redundant as we already do linting with flake8 via pre-commit (also also contradictory as we use google docstrings there).

Reply:
sorry for late reply. This job will be removed once we have pre-commit.ci installed (#35).